### PR TITLE
Add movie ID to front of forum movie title helper

### LINKED ISSUE #761: Regression: [movie] BBCode module no longer includes movie ID
Example:

[movie]1234[/movie] used to print:

> [1234] GBA Metroid Fusion "100%" by BioSpark

Now it prints:

> GBA Metroid Fusion "100%" by BioSpark

The ID is useful as a reference point for e.g. tracking awards, or disambiguating older/newer movies, so I'd like to see it back.

### PR DESCRIPTION
**Pull request by TiKevin83**
fixes #761 
Can someone else test locally? I'm having create post issues again

### CODE REVIEW

**Comment by YoshiRulz** [comment]
Should note: this prepends the ID to the title in 2(?) places in the frontend, but the rest of the title is assembled and saved to the database when initially creating the publication.

**Comment by adelikat** [comment]
> Should note: this prepends the ID to the title in 2(?) places in the frontend, but the rest of the title is assembled and saved to the database when initially creating the publication.

Yes.  The database value is a denormalization that is used in many places.  In those other places, the behavior here is not desired.

File: TASVideos.Core/Services/ForumWriterHelper.cs [hunk 1]
```diff
@@ -14,7 +14,16 @@ public ForumWriterHelper(ApplicationDbContext db)
 			_db = db;
 		}
 
-		public async Task<string?> GetMovieTitle(int id) => (await _db.Publications.FirstOrDefaultAsync(p => p.Id == id))?.Title;
+		public async Task<string?> GetMovieTitle(int id)
+		{
+			var publication = await _db.Publications.FirstOrDefaultAsync(p => p.Id == id);
+			if (publication is not null)
```
Comment (adelikat) [inline TASVideos.Core/Services/ForumWriterHelper.cs:20]: Can I ask you to make this a ternary? And if not, add a space after the brace, it is a stylecop violation which will then show up on other PRs.  Annoying that it doesn't show on PRs BEFORE it is merged :(

**Comment by nattthebear** [comment]
Are any changes desired to `GetSubmissionTitle`?  Did it used to show `[1234S]` in front of the name?

**Comment by TiKevin83** [comment]
> Are any changes desired to `GetSubmissionTitle`? Did it used to show `[1234S]` in front of the name?

It didn't in other places in the old site eg in movie lists

**Comment by Masterjun3** [comment]
It used to show `#1234:` for submission BBCode, and I think it already has the proper implementation, since it's part of the Title stored in the database.

File: TASVideos.Core/Services/ForumWriterHelper.cs (hunk 1 above)
Comment (TiKevin83) [reply TASVideos.Core/Services/ForumWriterHelper.cs]: updated per request

**Pull request merged**